packages/provider_devtools_extension/lib/src/provider_screen.dart

```diff
@@ -42,71 +46,97 @@ class ProviderScreenBody extends ConsumerWidget {
-                    },
+    // This change needed for extension to work in simulated environment
+    // since new option `requiresConnection` to config.yml
+    // not works in this config and it is needed to check connection
+    // inside plugin itself.
+    return ref.watch(serviceProvider).when(
+        data: (_) => SplitPane(
+              axis: splitAxis,
+              initialFractions: const [0.33, 0.67],
+              children: [
+                const RoundedOutlinedBorder(
+                  clip: true,
+                  child: Column(
+                    children: [
+                      AreaPaneHeader(
+                        roundedTopBorder: false,
+                        includeTopBorder: false,
+                        title: Text('Providers'),
+                      ),
+                      Expanded(
+                        child: ProviderList(),
+                      ),
+                    ],
                   ),
-                ],
-              ),
-              if (selectedProviderId != null)
-                Expanded(
-                  child: InstanceViewer(
-                    rootPath: InstancePath.fromProviderId(selectedProviderId),
-                    showInternalProperties: ref.watch(_showInternals),
+                ),
+                RoundedOutlinedBorder(
+                  clip: true,
+                  child: Column(
+                    children: [
+                      AreaPaneHeader(
+                        roundedTopBorder: false,
+                        includeTopBorder: false,
+                        title: Text(detailsTitleText),
+                        actions: [
+                          IconButton(
+                            icon: const Icon(Icons.settings),
+                            onPressed: () {
+                              unawaited(
+                                showDialog(
+                                  context: context,
+                                  builder: (_) =>
+                                      _StateInspectorSettingsDialog(),
+                                ),
+                              );
+                            },
+                          ),
+                        ],
+                      ),
+                      if (selectedProviderId != null)
+                        Expanded(
+                          child: InstanceViewer(
+                            rootPath:
+                                InstancePath.fromProviderId(selectedProviderId),
+                            showInternalProperties: ref.watch(_showInternals),
+                          ),
+                        ),
+                    ],
                   ),
                 ),
-            ],
-          ),
-        ),
-      ],
-    );
+              ],
+            ),
+        error: (_, stacktrace) => Center(
+              child: Text(
+                  'Error when connecting to VmService: \n${stacktrace.toString()}'),
+            ),
```

Comment:
I don't think there's value in showing that in the UI. It probably should be a snackbar or something.